### DISCUSSION TITLE
@StrainMeasure: Is it possible to obtain some intermediate results from the strain calculation

### DISCUSSION THREAD

**Discussion by MaynotbeGarychan** [thelfer/tfel#441] [Q&A] [answered]
Dear TFEL community,

This is Chen. Thanks for reading this message.
Recently, I am trying to implementing a UMAT of power-law J2 plasticity into AMITEX_FFTP using MFRONT to generate the code.
I use @StrainMeasure, for example, Hencky and GreenLagrangan. I can obtain the strain increment from "deto" in tensor form.
However, some variables, like the velocity gradient tensor, is also very important to me. In this case, would you please tell me that is it possible to obtain the velocity gradient tensor, and then I can stored them into StateVariable and then print it into .vtk.

Best,
Chen

**Comment by maxence-wz** [comment]
Hi @MaynotbeGarychan,
I think you can use @AuxiliaryStateVariable to define a variable seen as state variable from AMITEX. 
The `@AuxiliaryStateVariable` directive is used to declare internal variables (scalar or tensor) which are not part of the system to be integrated but which may be useful for post-processing. They must be updated with the @UpdateAuxiliaryStateVariables directive, which update the Auxiliary State Variables after the integration of internal variables (and the updating of stresses) but before external variables (temperature, total deformation).
Best regards,
Maxence

**Reply by MaynotbeGarychan** [reply]
Hi @maxence-wz,
Thanks for your reply. I am sorry my description to this question is not enough.
Now I am writing a UMAT for plasticity in finite strain framework. I just simply add the @StrainMeasure Hencky or GreenLagrangian into the code from https://tfel.sourceforge.net/IsotropicLinearHardeningPlasticity.html, to change this plasticity model from small strain theory to finite strain theory. The "deto" seems to be the strain increment, and the calculation from deformation gradient to strain increment would be conducted automatically if I add @StrainMeasure.
However, some intermediate results, like velocity gradient, which is calculated during the calculation of @StrainMeasure Hencky or GreenLagrangian, is very important to me. As this calculation is implicit in the mfront code, would you please tell me that is it possible to obtain these results like velocity gradient or deformation gradient?

Best wishes,
Chen

**Reply by thelfer** [reply, marked as answer]
@MaynotbeGarychan, At best you could compute an approximation of it using the push-forward of the green-lagrange strain rate (deto/dt) and something similar with the Hencky strain. 

Unfortunately, you  don't have access to deformation gradient to do this push forward operation, by design (there are good reasons to do this, too long to explain here).

The only solution is to compute this quantity on the `AMITEX_FFTP` side.

**Reply by MaynotbeGarychan** [reply]
@thelfer  Ok, I see. Thanks a lot for your anwser.
